src/components/timeline/TrackEventDelayGraph.js

```diff
@@ -0,0 +1,359 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+import * as React from 'react';
+import { withSize } from '../shared/WithSize';
+import explicitConnect from '../../utils/connect';
+import { formatMilliseconds } from '../../utils/format-numbers';
+import { getCommittedRange, getProfileInterval } from '../../selectors/profile';
+import { getThreadSelectors } from '../../selectors/per-thread';
+import { ORANGE_50 } from 'photon-colors';
+import Tooltip from '../tooltip/Tooltip';
+import EmptyThreadIndicator from './EmptyThreadIndicator';
+import bisection from 'bisection';
+
+import type { Thread, ThreadIndex } from '../../types/profile';
+import type { Milliseconds, CssPixels, StartEndRange } from '../../types/units';
+import type { SizeProps } from '../shared/WithSize';
+import type { ConnectedProps } from '../../utils/connect';
+
+import './TrackMemory.css';
+
+/**
+ * When adding properties to these props, please consider the comment above the component.
+ */
+type CanvasProps = {|
+  +thread: Thread,
+  +rangeStart: Milliseconds,
+  +rangeEnd: Milliseconds,
+  +interval: Milliseconds,
+  +width: CssPixels,
+  +height: CssPixels,
+  +lineWidth: CssPixels,
+  +eventDelays: Object, // TODO: add a real type for this object
+|};
+
+/**
+ * This component controls the rendering of the canvas. Every render call through
+ * React triggers a new canvas render. Because of this, it's important to only pass
+ * in the props that are needed for the canvas draw call.
+ */
+class TrackEventDelayCanvas extends React.PureComponent<CanvasProps> {
+  _canvas: null | HTMLCanvasElement = null;
+  _requestedAnimationFrame: boolean = false;
+
+  drawCanvas(canvas: HTMLCanvasElement): void {
+    const {
+      rangeStart,
+      rangeEnd,
+      height,
+      width,
+      lineWidth,
+      interval,
+      thread,
+      eventDelays,
+    } = this.props;
+    if (width === 0) {
+      // This is attempting to draw before the canvas was laid out.
+      return;
+    }
+
+    const ctx = canvas.getContext('2d');
+    const devicePixelRatio = window.devicePixelRatio;
+    const deviceWidth = width * devicePixelRatio;
+    const deviceHeight = height * devicePixelRatio;
```

Comment:
I'm not sure about the gradient for accessibility reasons but I'm not an expert on that area. Yeah, probably a horizontal line would be easier, but I would still prefer to make it as a follow-up because code is not technically wrong currently, and the sooner we land is the better since we've waited for so long already 😄  Created #2681 for that.